src/main/java/com/avail/interpreter/levelTwo/operand/L2ReadVectorOperand.java

```diff
@@ -112,72 +122,80 @@ public L2ReadVectorOperand (
 	 */
 	public List<R> registers ()
 	{
-		final List<R> registers = new ArrayList<>(elements.size());
-		for (final RR rr : elements)
-		{
-			registers.add(rr.register());
-		}
-		return registers;
+		return elements.stream()
+			.map(L2ReadOperand::register)
+			.collect(toList());
 	}
 
 	@Override
 	public abstract void dispatchOperand (final L2OperandDispatcher dispatcher);
 
 	@Override
 	public void instructionWasAdded (
-		final L2Instruction instruction,
 		final L2ValueManifest manifest)
 	{
-		for (final RR element : elements)
-		{
-			element.instructionWasAdded(instruction, manifest);
-		}
+		super.instructionWasAdded(manifest);
+		elements.forEach(
+			element -> element.instructionWasAdded(manifest));
 	}
 
 	@Override
-	public void instructionWasInserted (
-		final L2Instruction instruction,
+	public L2ReadVectorOperand<RR, R> adjustedForReinsertion (
 		final L2ValueManifest manifest)
 	{
+		final List<RR> newElements = new ArrayList<>(elements.size());
 		for (final RR element : elements)
 		{
-			element.instructionWasInserted(instruction, manifest);
+			final RR newElement = Casts.<L2ReadOperand<?>, RR>cast(
+				element.adjustedForReinsertion(manifest));
+			newElements.add(newElement);
 		}
```

Comment:
Because of the complexity of the loop body, the stream/map version would have to contain a lambda just like the for(){} version.  Not much value in that.